src/sqlancer/mysql/gen/MySQLSetGenerator.java

```diff
@@ -188,4 +193,49 @@ private SQLQueryAdapter get() {
         return new SQLQueryAdapter(sb.toString());
     }
 
+    public static SQLQueryAdapter resetOptimizer() {
+        return new SQLQueryAdapter("SET optimizer_switch='default'");
+    }
+
+    public static List<SQLQueryAdapter> getAllOptimizer(MySQLGlobalState globalState) {
+        List<SQLQueryAdapter> result = new ArrayList<>();
+        String[] options = { "index_merge", "index_merge_union", "index_merge_sort_union", "index_merge_intersection",
+                "engine_condition_pushdown", "index_condition_pushdown", "mrr", "mrr_cost_based", "block_nested_loop",
+                "batched_key_access", "materialization", "semijoin", "loosescan", "firstmatch", "duplicateweedout",
+                "subquery_materialization_cost_based", "use_index_extensions", "condition_fanout_filter",
+                "derived_merge", "use_invisible_indexes", "skip_scan", "hash_join", "subquery_to_derived",
+                "prefer_ordering_index", "derived_condition_pushdown" };
+
+        List<String> availableOptions = new ArrayList<>(Arrays.asList(options));
+        if (MySQLBugs.bug112242) {
+            availableOptions.remove("use_invisible_indexes");
+        }
+        if (MySQLBugs.bug112243) {
+            availableOptions.remove("subquery_to_derived");
+        }
+        if (MySQLBugs.bug112264) {
+            availableOptions.remove("block_nested_loop");
+        }
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("SET ");
+        if (globalState.getOptions().getNumberConcurrentThreads() == 1 && Randomly.getBoolean()) {
+            sb.append("GLOBAL");
+        } else {
+            sb.append("SESSION");
+        }
+        sb.append(" ");
```

Comment:
Sorry just noticed this, but I think we can just have a single method call here? `" optimizer_switch = '%s'"`? I'm surprised that PMD did not recognize this. I remember we had a rule that checked this automatically.

Reply:
If it's a hassle, we can merge it also without this change (in this case, feel free to hit the merge button), although it would be nice to fix it.